src/store/user.js

```diff
@@ -0,0 +1,152 @@
+import { userService } from '../services';
+import router from '../router';
+
+const token = localStorage.getItem('token');
+
+const namespaced = true;
+
+const state = token
+  ? { status: { loggedIn: true }, user: { token } }
+  : { status: {}, user: null };
+
+/* Actions */
+const login = ({ dispatch, commit }, { email, password, path }) => {
+  commit('loginRequest', { email });
+
+  userService.login(email, password)
+    .then(
+      (loginToken) => {
+        commit('loginSuccess', loginToken);
+        router.replace(path || '/conversations');
+      },
+      (error) => {
+        commit('loginFailure', error);
+        dispatch('alert/error', error, { root: true });
+      },
+    )
+    .catch((error) => {
+      commit('loginFailure', error);
+      dispatch('alert/error', error, { root: true });
+    });
```

Comment:
I think having both of these might be redundant?

Reply:
Oh true, I put both of them in case I wanted to handle thrown errors differently from rejections. But it makes sense in this case, to handle them in the same way right? Like to just set the alert error?

Reply:
Just doing `.catch()` should handle all exceptions or rejections I believe. [source](https://javascript.info/promise-error-handling#implicit-try-catch)

Reply:
Yeah it does, but what I meant in my comment was that putting the specific rejection handler in the `then`, lets us handle rejections differently from exceptions. Since doing just `.catch()` wouldn't tell us if the error was thrown or there was a rejection.

Reply:
I see. Yeah, I don't think we need both, at least for now. I'd just leave in the `catch()`.

Reply:
Kk will do.